slamd/materials/processing/models/material.py

```diff
@@ -1,20 +1,21 @@
-import uuid
-
-
-class Material:
-
-    def __init__(self, name='', type='', costs=None, additional_properties=None, is_blended=False):
-        self.uuid = uuid.uuid1()
-        self.name = name
-        self.type = type
-        self.costs = costs
-        self.additional_properties = additional_properties
-        self.is_blended = is_blended
+from uuid import UUID, uuid1
+from dataclasses import dataclass, field
+from slamd.materials.processing.models.additional_property import AdditionalProperty
 
 
+@dataclass
 class Costs:
+    co2_footprint: float = 0
+    costs: float = 0
+    delivery_time: int = 0
+
 
-    def __init__(self, co2_footprint=0, costs=0, delivery_time=0):
-        self.delivery_time = delivery_time
-        self.costs = costs
-        self.co2_footprint = co2_footprint
+@dataclass
+class Material:
+    # Generate a new UUID for every material, not one for every material
+    uuid: UUID = field(default_factory=lambda: uuid1())
+    name: str = ''
+    type: str = ''
+    costs: Costs = None
+    additional_properties: list[AdditionalProperty] = None
```

Comment:
we decided not to use typehints. Here you included it. I know that in principle it could be helpful, but this way it is inconsistent. If we want to have it only at dedicated places, it should be documented and there should be a concept of when to use it.